### PR TITLE
Fix relative path for .dlls in publish directory

### LINKED ISSUE #12386: [.NET 6] Files copies into MonoBundle get folder structure flattened
### Steps to Reproduce

1. Build example project using `dotnet build`

### Expected Behavior

`MailClient.Import.Wab.resources.dll` should be located in `./bin/Debug/net6.0-macos/osx-x64/macos-debug.app/Contents/MonoBundle/cs`

### Actual Behavior

`MailClient.Import.Wab.resources.dll` is located in `./bin/Debug/net6.0-macos/osx-x64/macos-debug.app/Contents/MonoBundle` (path component is lost)

### Environment

.NET 6 Preview 6

### Build Logs

[msbuild.binlog.zip](https://github.com/xamarin/xamarin-macios/files/6953282/msbuild.binlog.zip)

### Example Project (If Possible)

[Archive.zip](https://github.com/xamarin/xamarin-macios/files/6953278/Archive.zip)

### PR DESCRIPTION
**Pull request by filipnavara**
Fixes #12386 (and likely #12418)

### CODE REVIEW

File: dotnet/targets/Xamarin.Shared.Sdk.targets [hunk 1]
```diff
@@ -973,7 +973,7 @@
 			<!-- include .dll, .exe and, for debugging only, .pdb files inside the .app -->
 			<ResolvedFileToPublish
 				Update="@(ResolvedFileToPublish)"
-				RelativePath="$(_AssemblyPublishDir)\%(ResolvedFileToPublish.DestinationSubDirectory)\%(Filename)%(Extension)"
+				RelativePath="$(_AssemblyPublishDir)\%(ResolvedFileToPublish.RelativePath)"
```
Comment (spouliot) [inline dotnet/targets/Xamarin.Shared.Sdk.targets:976]: c.c. @rolfbjarne 

https://github.com/xamarin/xamarin-macios/pull/12333/files#diff-2e24f1b6f44df7580244d00f106fa1a9e2257536fbca9d7bbdc74571549324a9L952

I'm surprised this did not trigger any of our existing tests.

**Review by spouliot — COMMENTED** [review]
seems incomplete
and we'll need some tests

File: dotnet/targets/Xamarin.Shared.Sdk.targets (hunk 1 above)
Comment (rolfbjarne) [reply dotnet/targets/Xamarin.Shared.Sdk.targets:976]: Yes, this has been discussed a bit here: https://github.com/xamarin/xamarin-macios/issues/12386#issuecomment-898562065

For normal project references our logic works just fine, but the test case is setting `CopyToOutputDirectory=PreserveNewest` on `None` items, which is something we don't have tests for.

**Comment by filipnavara** [comment]
Closing in favor of https://github.com/rolfbjarne/xamarin-macios/commits/issue-12386-bundlestructure-bundleresource-content

**Pull request closed without merging**